sdks/python/apache_beam/io/watermark_estimators_test.py

```diff
@@ -0,0 +1,105 @@
+#
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Unit tests for built-in WatermarkEstiamtors"""
+
+from __future__ import absolute_import
+
+import unittest
+
+from apache_beam.io.iobase import WatermarkEstimator
+from apache_beam.io.watermark_estimators import ManualWatermarkEstimator
+from apache_beam.io.watermark_estimators import MonotonicWatermarkEstimator
+from apache_beam.io.watermark_estimators import WalltimeWatermarkEstimator
+from apache_beam.utils.timestamp import Duration
+from apache_beam.utils.timestamp import Timestamp
+
+
+class MonotonicWatermarkEstimatorTest(unittest.TestCase):
+
+  def test_initialize_from_state(self):
+    timestamp = Timestamp(10)
+    watermark_estimator = MonotonicWatermarkEstimator(timestamp)
+    self.assertIsInstance(watermark_estimator, WatermarkEstimator)
+    self.assertEqual(watermark_estimator.get_estimator_state(), timestamp)
+
+  def test_observe_timestamp(self):
+    watermark_estimator = MonotonicWatermarkEstimator(Timestamp(10))
+    watermark_estimator.observe_timestamp(Timestamp(15))
+    self.assertEqual(watermark_estimator.current_watermark(), Timestamp(15))
+    watermark_estimator.observe_timestamp(Timestamp(20))
+    self.assertEqual(watermark_estimator.current_watermark(), Timestamp(20))
+    watermark_estimator.observe_timestamp(Timestamp(20))
+    self.assertEqual(watermark_estimator.current_watermark(), Timestamp(20))
+    with self.assertRaises(ValueError):
+      watermark_estimator.observe_timestamp(Timestamp(10))
+
+  def test_get_estimator_state(self):
+    watermark_estimator = MonotonicWatermarkEstimator(Timestamp(10))
+    watermark_estimator.observe_timestamp(Timestamp(15))
+    self.assertEqual(watermark_estimator.get_estimator_state(), Timestamp(15))
+
+
+class WalltimeWatermarkEstimatorTest(unittest.TestCase):
+
+  def test_initialization(self):
+    watermark_estiamtor = WalltimeWatermarkEstimator()
+    self.assertIsInstance(watermark_estiamtor, WatermarkEstimator)
+    self.assertIsNone(watermark_estiamtor.get_estimator_state())
+
+  def test_observe_timestamp(self):
+    watermark_estiamtor = WalltimeWatermarkEstimator()
```

Comment:
You can test this explicitly if you can pass in a clock to the constructor, same for below.

Reply:
Seems like we don't use clock in python sdk. Instead, mock should work.